src/packages_push.go

```diff
@@ -0,0 +1,90 @@
+package src
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+)
+
+func PushPackagesForRepo(cacheDir, ghPatToken, sourceRepoName, destinationURL, destinationToken, destinationRepoName, ghAPIUrl string) error {
+
+	tags, err := ReadValidPackageTagsFromCache(cacheDir, sourceRepoName)
+	if err != nil {
+		return fmt.Errorf("Error getting list of tags for packages for repo %s: %s", sourceRepoName, err)
+	}
+
+	for _, tag := range tags {
+		err = PushPackageForTag(cacheDir, sourceRepoName, ghPatToken, destinationURL, destinationToken, destinationRepoName, tag, ghAPIUrl)
+		if err != nil {
+			return fmt.Errorf("Error pushing package for tag %s: %s", tag, err)
+		}
+	}
```

Comment:
Can we do this in parallel? are we exiting the complete process if it fails to push one package version ?

Reply:
Updated this.